src/App.jsx

```diff
@@ -14,45 +15,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [status, setStatus] = useState({
+    alphabetical: false,
+    bylength: false,
+    reversed: false,
+  });
+
+  const isModified = status.alphabetical || status.bylength || status.reversed;
+
+  function sortAlphabetically() {
+    setGoods((prevState) => {
+      if (!status.reversed) {
+        return [...prevState].sort();
+      }
+
+      return [...prevState].sort((a, b) => b.localeCompare(a));
+    });
+
+    setStatus(prevState => ({
+      ...prevState,
+      alphabetical: true,
+      bylength: false,
+    }));
+  }
+
+  function sortByLength() {
+    setGoods((prevState) => {
+      if (!status.reversed) {
+        return [...prevState].sort((a, b) => a.length - b.length);
+      }
+
+      return [...prevState].sort((a, b) => b.length - a.length);
+    });
+
+    setStatus(prevState => ({
+      ...prevState,
+      bylength: true,
+      alphabetical: false,
+    }));
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function reverseGoods() {
+    setGoods(prevState => [...prevState].reverse());
+    setStatus(prevState => ({
+      ...prevState,
+      reversed: !prevState.reversed,
+    }));
+  }
+
+  function resetGoods() {
+    setGoods(goodsFromServer);
+    setStatus({
+      alphabetical: false,
+      bylength: false,
+      reversed: false,
+    });
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            status.alphabetical ? 'button is-info' : 'button is-info is-light'
+          }
+          onClick={sortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            status.bylength ? 'button is-success' : 'button is-success is-light'
```

Comment:
use [classnames](https://www.npmjs.com/package/classnames) library to manage dynamic classnames (already exists in project dependencies).
```
import cn from 'classnames';
```
```suggestion
          className={cn('button is-success', { 'is-light': status.sort === 'length' } )
```